docs/testplan/BGP-Convergence-Testplan-for-Benchmark-Performance.md

```diff
@@ -0,0 +1,190 @@
+# BGP convergence test plan for benchmark performance
+
+- [BGP convergence test plan for benchmark performance](#bgp-convergence-test-plan-for-benchmark-performance)
+  - [Overview](#Overview)
+    - [Scope](#Scope)
+    - [Testbed](#Keysight-Testbed)
+  - [Topology](#Topology)
+    - [SONiC switch as ToR](#SONiC-switch-as-ToR)
+    - [SONiC switch as Leaf](#SONiC-switch-as-Leaf)
+  - [Setup configuration](#Setup-configuration)
+  - [Test methodology](#Test-methodology)
+  - [Test cases](#Test-cases)
+    - [Test case # 1 – BGP Remote Link Failover Convergence (route withdraw)](#test-case--1--convergence-performance-when-remote-link-fails-route-withdraw)
+      - [Test objective](#Test-objective)
+      - [Test steps](#Test-steps)
+      - [Test results](#Test-results)
+      - [Test case](#Test-case)
+    - [Test case # 2 – BGP RIB-IN Convergence](#Test-case--2--RIB-IN-Convergence)
+      - [Test objective](#Test-objective-1)
+      - [Test steps](#Test-steps-1)
+      - [Test results](#Test-results-1)
+      - [Test case](#Test-case-1)
+    - [Test case # 3 - BGP Local Link Failover Convergence](#Test-case--3--Failover-convergence-with-local-link-failure) 
+      - [Test objective](#Test-objective-2)
+      - [Test steps](#Test-steps-2)
+      - [Test results](#Test-results-2)
+      - [Test case](#Test-case-2)
+    - [Call for action](#Call-for-action)
+
+## Overview
+The purpose of these tests is to test the overall convergence of a data center network by simulating multiple network devices such as ToR/Leafs and using SONiC switch DUT as one of the ToR/Leaf, closely resembling production environment.
+
+### Scope
+These tests are targeted on fully functioning SONiC system. The purpose of these tests are to measure convergence when some unexpected failures such as remote link failure, local link failure, node failure or link faults etc occur and some expected failures such as maintenance or upgrade of devices occur in the SONiC system.
+
+### Keysight Testbed
+The tests will run on following testbeds:
+* t0
+
+![Single DUT Topology ](Img/Single_DUT_Topology.png)
+
+## Topology
+### SONiC switch as ToR
+
+![SONiC DUT as ToR ](Img/Switch_as_ToR.png)
+
+### SONiC switch as Leaf
+
+![SONiC DUT as ToR ](Img/Switch_acting_as_leaf.png)
+
+## Setup configuration
+IPv4 EBGP neighborship will be configured between SONiC DUT and directly connected test ports. Test ports inturn will simulate the ToR's and Leafs by advertising IPv4/IPv6, dual-stack routes.
```

Comment:
How about BGP sessions between keysight emulated switches?

Reply:
We are using eBGP peering between the DUT and the spine, in order to simulate the emulated leaf switches we are appending the AS-PATHS to the routes which are being advertised.